src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -330,7 +341,38 @@ def _get_pool_status(
                 });
 
         return (ret, root_map, pool_root)
+    
+    def _update_progress_events(self):
+        osdmap = self.get_osdmap()
+        pools = osdmap.get_pools()
+        for pool_id in list(self._event):
+            ev = self._event[pool_id]
+            pool_data = pools[int(pool_id)]
+            pg_num = pool_data['pg_num']
+            pg_num_target = pool_data['pg_num_target']
+            initial_pg_num = ev._pg_num
+            initial_pg_num_target = ev._pg_num_target
+            progress = (pg_num - initial_pg_num)/(pg_num_target - initial_pg_num)
+            if pg_num == pg_num_target:
+                self.remote('progress', 'complete', ev._ev_id)
+                del self._event[pool_id]
+                continue
+            elif pg_num == initial_pg_num:
+                continue
+
+            elif pg_num_target > pg_num:
```

Comment:
these two cases only differ by one word now, could make 'increasing'/'decreasing' another variable in the message to avoid repitition